app/uk/gov/hmrc/rasapi/repository/RasFileRepository.scala

```diff
@@ -15,116 +15,115 @@
-    log.debug(s"Checking if file exists $fileId ")
-    gridFSG.find[BSONDocument, ResultsFile](BSONDocument("_id" -> fileId)).headOption.recover{
-      case ex:Throwable =>
-        log.error(s"error trying to find if parent file record Exists $fileId for . Exception: ${ex.getMessage}")
-        throw new RuntimeException("failed to check file exists due to error" + ex.getMessage)
+  def fetchResultsFile(fileId: ObjectId): Future[Option[ResultsFile]] = {
+    log.debug(s"[RasFileRepository][isFileExists] Checking if file exists with id: $fileId ")
+    gridFSG.find(Document("_id" -> fileId)).headOption.recover {
+      case ex: Throwable =>
+        log.error(s"[RasFileRepository][isFileExists] Error trying to find if parent file record exists for id: $fileId. Exception: ${ex.getMessage}")
+        throw new RuntimeException("Failed to check file exists due to error ${ex.getMessage}")
     }
   }
 
-  private def getBsonObjectIdFromFileName(filename: String): Future[Option[BSONObjectID]] =
-    gridFSG.files.find(BSONDocument("filename" -> filename), Some(BSONDocument("_id" -> 1))).one[BSONDocument].map {
-      optionalDocument => optionalDocument.flatMap { document =>
-        document.getAs[BSONObjectID]("_id")
-      }
-    }
-
   def removeFile(fileName: String, userId: String): Future[Boolean] = {
-    log.debug(s"file to remove => fileName: $fileName for userId ($userId).")
-    getBsonObjectIdFromFileName(fileName).flatMap {
-      case Some(bsonObjectId) =>
-        log.info(s"[RasFileRepository][removeFile] successfully got id for file. BSONObjectId: ${bsonObjectId.stringify}")
-        gridFSG.remove(bsonObjectId).map {
-          res =>
-            if(res.writeErrors.isEmpty){
-              log.info(s"Results file removed successfully for userId ($userId) with the file named $fileName")
-            } else {
-              log.error(s"error while removing file ${res.writeErrors.toString} for userId ($userId).")
-            }
-            res.writeErrors.isEmpty
+    log.debug(s"[RasFileRepository][removeFile] File to remove => fileName: $fileName for userId ($userId).")
+    gridFSG.find(Document("filename" -> fileName)).headOption.flatMap {
+      case file @Some(_) =>
+        val objectId: ObjectId = file.get.getObjectId
+        log.info(s"[RasFileRepository][removeFile] Successfully retrieved ObjectId: $objectId")
+        log.info(s"[RasFileRepository][removeFile] Attempting to delete file with ObjectId: $objectId")
+        Await.result(gridFSG.delete(objectId).toFuture(), Duration(10, SECONDS))
+        fetchResultsFile(objectId).map {
+          case file @Some(_) =>
+            log.error(s"[RasFileRepository][removeFile] Error while removing file ${file.get.getFilename} for userId ($userId).")
+            false
+          case None =>
+            log.info(s"[RasFileRepository][removeFile] Results file removed successfully for userId ($userId) with the file named $fileName")
+            true
```

Comment:
this could be cleaned up with some for-comprehension, e.g.
```
for {
  _ <- gridFSG.delete(objectId).toFuture()
  maybeFile <- fetchResultsFile(objectId)
} yield {
  maybeFile match {
    case Some(file) => 
       ...
       false
     case None =>
       ...
       true
  }
}
```